Source/JavaScriptCore/builtins/TypedArrayConstructor.js

```diff
@@ -64,6 +65,17 @@ function from(items /* [ , mapfn [ , thisArg ] ] */)
         if (!@isCallable(iteratorMethod))
             @throwTypeError("TypedArray.from requires that the property of the first argument, items[Symbol.iterator], when exists, be a function");
 
+        // If iterator-method is TypedArray#values, then it is ensured that we can iterate |arrayLike| without side effect
+        // since TypedArray does not accept any accessor definitions for indexed properties.
+        // We also need to ensure that |this| constructor will not cause user observable side-effect. Otherwise, it is possible
+        // that |this| constructor detaches arrayLike before using.
+        if (!mapFn && @isTypedArrayView(arrayLike) && iteratorMethod === @typedArrayValues && @isTypedArrayViewConstructor(this)) {
```

Comment:
Oops, I think I need a bit more checks here. Let's consider that we override iterator's prototype.

```
%ArrayIterator%.prototype.next = function () { return { value: 42, done: false }; };
```
Then, we should use this function to iterate typed array to collect values.
This means that we need to ensure that array-iterator-protocol is not broken here. `isArrayPrototypeIteratorProtocolFastAndNonObservable` should be checked.
I'll extend this patch to add this and test (After that, I'll request another review).

Reply:
I've added `canUseFastTypedArrayFrom` function.

Reply:
Ugh, we still need to validate a bit more: we should check whether this `arrayLike` does not have `length` / index own property. Fix forthcoming.

Reply:
We need TypedArray iterator protocol validation. Because,

1. We can still put `length` value / getter in own typed array or `%TypedArray%.prototype`.
2. On the other hand, indexed access cannot be trapped & overridden

We should have sane chain watchpoint for typed array prototypes. And we should also have a watchpoint whether `length` gets overridden in prototypes.
Also, we need to check that we do not have `length` own property on `arrayLike`.